File: Go/src/Go/Bubble.go
package main

import (
	"fmt"
	"github.com/pkg/profile"
	"io/ioutil"
	"os"
	"strconv"
	"strings"
	"time"
)

func Bubble(numbers[] int)[]int {

	var temp = 0;

	for i := 0; i < len(numbers); i++ {

		for j := 0; j < len(numbers) - 1; j++ {

			if numbers[j] > numbers[j + 1]{

				temp = numbers[j]

				numbers[j] = numbers[j + 1]

				numbers[j + 1] = temp

			}

		}

	}

	return numbers

}

func main() {

	dir, _ := os.Getwd()

	defer profile.Start(profile.MemProfile, profile.MemProfileRate(100), profile.ProfilePath(dir)).Stop()

	for i := 0; i < 10000; i++ {

		tmp := make([]byte, 100000)

		tmp[0] = tmp[1] << 0

	}

	// b, _ := ioutil.ReadFile("/Users/henrique_buzin/go/src/Go/100.000.txt")
	// b, _ := ioutil.ReadFile("/Users/henrique_buzin/go/src/Go/100.000.txt")
	b, _ := ioutil.ReadFile("/Users/henrique_buzin/go/src/Go/100.000.txt")

	var numbers []int

	var lines = strings.Split(string(b), "\n")

	for i := 0; i < len(lines); i++ {

		number, _ := strconv.Atoi(lines[i])

		numbers = append(numbers, number)

	}

	var inicio = time.Now().UnixNano()

	Bubble(numbers)

	var fim = time.Now().UnixNano()

	fmt.Println(numbers)

	fmt.Println(fim - inicio)

}

File: Scala/src/main/scala/Bubble.scala
import scala.io.Source

object Bubble {

  def Bubble(numbers: Array[Int]): Array[Int] = {

    var temp = 0

    for (i <- 0 until numbers.length) {

      for (j <- 0 until numbers.length - 1) {

        if(numbers(j) > numbers(j + 1)){

          temp = numbers(j)

          numbers(j) = numbers(j + 1)

          numbers(j + 1) = temp

        }

      }

    }

    numbers

  }

  def main(args: Array[String]){

    val numbers = new Array[Int](1000)

    // for ((number, i) <- Source.fromFile("/Users/henrique_buzin/IdeaProjects/Scala/src/main/scala/100.000.txt").getLines.zipWithIndex) {
    // for ((number, i) <- Source.fromFile("/Users/henrique_buzin/IdeaProjects/Scala/src/main/scala/10.000.txt").getLines.zipWithIndex) {
    for ((number, i) <- Source.fromFile("/Users/henrique_buzin/IdeaProjects/Scala/src/main/scala/1.000.txt").getLines.zipWithIndex) {

      numbers(i) = number.toInt

    }

    val inicio = System.nanoTime()

    Bubble(numbers)

    val fim = System.nanoTime()

    println(numbers.deep)

    println(fim - inicio)

    println(Runtime.getRuntime.totalMemory - Runtime.getRuntime.freeMemory)

  }

}